wikimedia/wikipedia-ios
# Fix for Norwegian appearing twice when upgrading from 6.7.4

**Fixes Phabricator ticket:** 
https://phabricator.wikimedia.org/T281378

### Notes
Add a data library migration to ensure preferred language codes are unique:
- Add a method to MWKLanguageLinkController to unique preferred language codes
- Add test to ensure the method of uniquing behaves as expected
- Add new library version and migration to MWKDataStore

### Test Steps
1. Check out releases/6.7.4 on a device with only Norwegian Bokmal as a language.
2. Launch app, go through onboarding. The language page will be in that weird empty state due to a 6.7.4 bug. Add Norsk as a language.

I tested two different paths:
3. Checkout releases/6.8.0, build and run so the problem manifests itself on 6.8
4. Build and run this code, where the extra listing of Norwegian is taken care of.

Then repeat steps 1 & 2 testing going directly from 6.7.4 to 6.8.1 with this fix:
3. Build and run this code, where the extra listing of Norwegian is taken care of.
